File: 프로그래머스/4/42891. 무지의 먹방 라이브/무지의 먹방 라이브.java
import java.util.*;

class Solution {
    public int solution(int[] food_times, long k) {
        
        int n = food_times.length;
        List<int[]> list = new ArrayList<>();
        
        for(int i=0; i<food_times.length; i++){
            list.add(new int[]{food_times[i], i+1});
        }
        
        list.sort((a, b) -> a[0]-b[0]);
        
        long prev = 0;
        long remain = n;
        
        for(int i=0; i<n; i++){
            long now = list.get(i)[0];
            long diff = now-prev;
            if(diff != 0){
                long del = diff * remain;
                if(k >= del){
                    k -= del;
                    prev = now;
                }else{
                    List<int[]> rest = list.subList(i, n);
                    rest.sort((a, b) -> a[1] - b[1]);
                    int answer = rest.get((int)(k%remain))[1];
                    return answer;
                }
            }
            remain--;
        }
        
        return -1;
    }
}

File: 프로그래머스/4/42891. 무지의 먹방 라이브/무지의 먹방 라이브.js
function solution(food_times, k) {
    
    var n = food_times.length;
    const list = [];
    
    for(let i=0; i<n; i++){
        list.push([food_times[i], i+1]);
    }
    
    list.sort((a,b) => a[0]-b[0]);
    
    var prev = 0;
    var remain = n;
    
    for(let i=0; i<n; i++){
        let now = list[i][0];
        let diff = now-prev;
        if(diff != 0){
            let del = diff*remain;
            if(k >= del){
                k -= del;
                prev = now;
            }else{
                const rest = list.slice(i);
                rest.sort((a,b) => a[1]-b[1]);
                return rest[k%remain][1];
            }
        }
        remain --;
    }
    return -1;
}